scripts/policy/frameworks/management/persistence.zeek

```diff
@@ -0,0 +1,52 @@
+##! Common adjustments for any kind of Zeek node when we run the Management
+##! framework.
+
+@load base/misc/installation
+@load base/utils/paths
+
+@load ./config
+
+# For testing, keep persistent state local to the current working directory,
+# and disable log rotation.
+@if ( getenv("ZEEK_MANAGEMENT_TESTING") != "" )
+
+redef Management::spool_dir = ".";
+redef Management::state_dir = ".";
+redef Log::default_rotation_interval = 0 secs;
+
+@else
+
+# For any kind of Zeek process we steer rotated logs awaiting archival into a
+# queue directory in the spool.
+redef Log::default_rotation_dir = build_path(Management::get_spool_dir(), "log-queue");
+
+# Additional default storage locations:
+redef Broker::table_store_db_directory = build_path(Management::get_state_dir(), "broker");
+@if ( getenv("CLUSTER_NODE") != "" )
+redef Cluster::default_store_dir = build_path(Management::get_state_dir(), "cluster");
+@endif
```

Comment:
This is a central directory for all nodes, right? I'm wondering whether the storage backing a table should stay node-specific. In principle one can create Broker storage masters on any node. If that's useful, might depend; but if one does, I don't think they their storage should be visible to each other (because you'd be using clones if you want state synced across).

Reply:
Yeah, it is. Here too, I was wondering too. :-) This is another concept from zeekctl, per its `BrokerDBDir` and `DefaultStoreDir` options, which are also global. If we can do without these, I'd prefer it.

It does bring up one question: where should each node's directory live? Right now that's under `spool/nodes`, with the assumption that there's only log content in there. That's already a bit of a stretch given that the `stdout`/`stderr` files are in there too. My take on spools is that one thing dumps stuff there, and another thing cleans it out, so for non-logs that doesn't fit well. With this change to the stores, I'm thinking the node directories should instead be `var/lib/nodes/<name>`. The spool would then really only be for rotated but not yet archived logs — perhaps in `spool/logs/`, to leave room for other types of spool content in the future.

Does that sound good?

Reply:
Yes, that sounds good. That indeed expresses the difference in semantics as well: this is important data that's to be persisted, rather a "spool" where things just move through.